lib/model/store.dart

```diff
@@ -850,8 +916,11 @@ class UpdateMachine {
     BackoffMachine? backoffMachine;
 
     while (true) {
+      if (_disposed) return;
+
       if (_debugLoopSignal != null) {
```

Comment:
(followup on https://github.com/zulip/zulip-flutter/pull/948#discussion_r1805674256 )

```suggestion
    assert(!_disposed);

    BackoffMachine? backoffMachine;
    while (true) {
      if (_debugLoopSignal != null) {
```

I think a check at the top of the loop shouldn't be needed — it's not in itself right after an await, and we already have checks after each await.

Conversely an assert at the top of the whole method is good for the same reason as on other methods: it helps us catch if we're attempting to call the method when the object is already disposed, which would be a bug.

Reply:
Oh indeed! Reading more closely, I see you did suggest putting it at the top of the method, not the top of the loop.